### PR TITLE
Modify and split CompactorMain into logical classes

### PR DESCRIPTION
**Pull request by SravanthiAshokKumar**
Split CorfuStoreCompactorMain into CorfuCompactorCheckpointer and CorfuCompactorControls
-The CorfuCompactorControls class holds all the controls related paramters and 
performs required operation accordingly
-The CorfuCompactorCheckpointer starts checkpointing tables if there is an ongoing compaction cycle

## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: corfudb-tools/src/main/java/org/corfudb/compactor/CorfuCompactorCheckpointer.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import org.corfudb.runtime.DistributedCheckpointer;
+import org.corfudb.runtime.DistributedCheckpointerHelper;
+import org.corfudb.runtime.ServerTriggeredCheckpointer;
+import org.corfudb.runtime.collections.CorfuStore;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Invokes the startCheckpointing() method of DistributedCompactor to checkpoint tables that weren't
```
Comment (hisundar) [inline corfudb-tools/src/main/java/org/corfudb/compactor/CorfuCompactorCheckpointer.java:16]: this comment is outdated right?

File: corfudb-tools/src/main/java/org/corfudb/compactor/CorfuCompactorControlsConfig.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package org.corfudb.compactor;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Getter
+@Slf4j
+public class CorfuCompactorControlsConfig extends CorfuCompactorConfig {
```
Comment (hisundar) [inline corfudb-tools/src/main/java/org/corfudb/compactor/CorfuCompactorControlsConfig.java:8]: Maybe a comment for this class to say it is only parsing options here?

**Review by hisundar — APPROVED** [review]
lgtm

**Review by Lujie1996 — APPROVED** [review]
Good work!
Nit: I find the class names a little convoluted. Maybe call them `CompactorCheckpointer, CompactorController, CompactorBaseConfig, CompactorControllerConfig`?

**Pull request merged**